### PR TITLE
Feature/create new env flow

### PR DESCRIPTION
**Pull request by opoliarush**
Impl #747

### CODE REVIEW

File: src/components/RecentActivity/CreateEnvironmentWizard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  };
+
+  const getStepVisibility = (key: string) => {
+    const stepIndex = getSteps().findIndex((x) => x.key == key);
+
+    if (stepIndex === currentStep) {
+      return true;
+    }
+
+    return false;
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/index.tsx:179]: simplify

File: src/components/RecentActivity/CreateEnvironmentWizard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        ) : (
+          <s.Step key="finish" $isVisible={completed}>
+            <EnvironmentCreated
+              goToEnvironment={() => {
+                onClose(newEnvironment.id);
+              }}
+            />
+          </s.Step>
+        )}
+        {!!errors.length && <ErrorsPanel errors={errors} />}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/index.tsx:251]:
```suggestion
        {errors.length > 0 && <ErrorsPanel errors={errors} />}
```

File: src/components/RecentActivity/EnvironmentInstructionsPanel/index.tsx [hunk 3]
```diff
@@ -147,7 +149,7 @@ curl --create-dirs -O -L --output-dir ./otel https://github.com/digma-ai/otel-ja
 export JAVA_TOOL_OPTIONS="-javaagent:/otel/opentelemetry-javaagent.jar -Dotel.exporter.otlp.endpoint=http://${hostname}:5050 -Dotel.javaagent.extensions=/otel/digma-otel-agent-extension.jar -Dotel.metrics.exporter=none -Dotel.logs.exporter=none -Dotel.exporter.otlp.protocol=grpc"
 
 export OTEL_SERVICE_NAME={--ENTER YOUR SERVICE NAME HERE--}
-export OTEL_RESOURCE_ATTRIBUTES=digma.environment=${props.environment.originalName}`}
+export OTEL_RESOURCE_ATTRIBUTES=digma.environment=${props.environment.id}`}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/EnvironmentInstructionsPanel/index.tsx:152]:
```suggestion
export OTEL_RESOURCE_ATTRIBUTES=digma.environment.id=${props.environment.id}`}
```

File: src/components/common/icons/12px/CheckCircleIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        stroke={color}
+        strokeLinecap="round"
+        strokeLinejoin="round"
+        clipPath="url(#check-circle-1)"
+      >
+        <path d="M4.625 6.375 5.75 7.5l2.625-2.625" />
+        <path d="M6.5 10.5a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9Z" />
+      </g>
+      <defs>
+        <clipPath id="check-circle-1">
```
Comment (kshmidt-digma) [inline src/components/common/icons/12px/CheckCircleIcon.tsx:26]:
```suggestion
        clipPath="url(#check-circle-clip-1)"
      >
        <path d="M4.625 6.375 5.75 7.5l2.625-2.625" />
        <path d="M6.5 10.5a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9Z" />
      </g>
      <defs>
        <clipPath id="check-circle-clip-1">
```

File: src/components/common/icons/12px/ErrorIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      width={size}
+      height={size}
+      viewBox="0 0 12 12"
+      fill="none"
+    >
+      <g
+        stroke={color}
+        strokeLinecap="round"
+        strokeLinejoin="round"
+        clipPath="url(#error-icon)"
```
Comment (kshmidt-digma) [inline src/components/common/icons/12px/ErrorIcon.tsx:20]:
```suggestion
        clipPath="url(#error-icon-clip-1)"
```

File: src/components/common/icons/12px/ErrorIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      >
+        <path d="m7.5 4.5-3 3m0-3 3 3m-1.5 3a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9Z" />
+      </g>
+      <defs>
+        <clipPath id="#error-icon">
```
Comment (kshmidt-digma) [inline src/components/common/icons/12px/ErrorIcon.tsx:25]:
```suggestion
        <clipPath id="#error-icon-clip-1">
```

File: src/components/common/icons/16px/ErrorIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      width={size}
+      height={size}
+      viewBox="0 0 16 16"
+      fill="none"
+    >
+      <g
+        stroke={color}
+        strokeLinecap="round"
+        strokeLinejoin="round"
+        clipPath="url(#error-icon)"
```
Comment (kshmidt-digma) [inline src/components/common/icons/16px/ErrorIcon.tsx:20]:
```suggestion
        clipPath="url(#error-icon-clip-1)"
```

File: src/components/common/icons/16px/ErrorIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      >
+        <path d="m10 6-4 4m0-4 4 4m-2 4A6 6 0 1 0 8 2a6 6 0 0 0 0 12Z" />
+      </g>
+      <defs>
+        <clipPath id="error-icon">
```
Comment (kshmidt-digma) [inline src/components/common/icons/16px/ErrorIcon.tsx:25]:
```suggestion
        <clipPath id="error-icon-clip-1">
```

File: src/components/common/icons/20px/CheckCircleIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      width={size}
+      height={size}
+      viewBox="0 0 20 20"
+      fill="none"
+    >
+      <g
+        stroke={color}
+        strokeLinecap="round"
+        strokeLinejoin="round"
+        clipPath="url(#check-circle-1)"
```
Comment (kshmidt-digma) [inline src/components/common/icons/20px/CheckCircleIcon.tsx:20]:
```suggestion
        clipPath="url(#check-circle-clip-1)"
```

File: src/components/common/icons/20px/CheckCircleIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      >
+        <path d="M6.875 10.625 8.75 12.5l4.375-4.375" />
+        <path d="M10 17.5a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15Z" />
+      </g>
+      <defs>
+        <clipPath id="check-circle-1">
```
Comment (kshmidt-digma) [inline src/components/common/icons/20px/CheckCircleIcon.tsx:26]:
```suggestion
        <clipPath id="check-circle-clip-1">
```

File: src/components/common/v3/TextField/styles.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+export const Container = styled.div<ContainerProps>`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  padding: 4px 6px;
+  border-radius: 4px;
+  border: 1px solid
+    ${({ theme, $focused, $isInValid }) => {
+      if ($isInValid) {
```
Comment (kshmidt-digma) [inline src/components/common/v3/TextField/styles.ts:13]:
```suggestion
      if ($isInvalid) {
```

**Review by kshmidt-digma — CHANGES_REQUESTED** [review]

File: src/components/Navigation/ScopeNavigation/index.tsx [hunk 12]
```diff
@@ -90,15 +83,6 @@ export const ScopeNavigation = (props: ScopeNavigationProps) => {
             }
           });
         }
-
-        if (historyStep && historyStep.environment) {
```
[the remark this answers is not in the document]
Comment (opoliarush) [reply src/components/Navigation/ScopeNavigation/index.tsx:94]: fixed

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentPanel/Tab/index.tsx [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+import { StepStatus } from "../../types";
+import * as s from "./styles";
+import { TabProps } from "./types";
+
+const getState = (index: number, theme: DefaultTheme, state: StepStatus) => {
+  switch (state) {
+    case "completed":
+      return <CheckIcon color={theme.colors.v3.status.success} />;
+    case "error":
+      return <ErrorIcon size={20} color={theme.colors.v3.status.high} />;
```
[the remark this answers is not in the document]
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentPanel/Tab/index.tsx:14]: fixed

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentCreated/index.tsx [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+import * as s from "./styles";
+import { EnvironmentCreatedProps } from "./types";
+
+export const EnvironmentCreated = ({
+  goToEnvironment
+}: EnvironmentCreatedProps) => {
+  const theme = useTheme();
+  return (
+    <s.Container>
+      <CheckCircleIcon size={20} color={theme.colors.v3.status.success} />
```
[the remark this answers is not in the document]
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentCreated/index.tsx:13]: done

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentNameStep/index.tsx [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    if (!ENVIRONMENT_NAME_REGEX.test(value)) {
+      setIsValid(false);
+      return;
+    }
+    setIsValid(true);
+    onNameChange(value);
+  };
+
+  const getInputState = () => {
+    if (name === null) {
```
[the remark this answers is not in the document]
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentNameStep/index.tsx:41]: done

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentPanel/index.tsx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+          {tabs.map((tab) => {
+            return (
+              <Tab
+                key={tab.index}
+                index={tab.index}
+                name={tab.name}
+                state={tab.state}
+              />
+            );
+          })}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentPanel/index.tsx:41]:
```suggestion
          {tabs.map((tab) => 
              <Tab
                key={tab.name}
                index={tab.index}
                name={tab.name}
                state={tab.state}
              />
         )}
```

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+
+type Story = StoryObj<typeof meta>;
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Default: Story = {
+  args: {
+    onClose: () => {
+      // close
+    }
+  }
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:27]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+};
+
+export const Errored: Story = {
+  args: {
+    onClose: () => {
+      // close
+    }
+  },
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:35]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    }, 1000);
+  }
+};
+
+export const WithoutRegisteredUser: Story = {
+  args: {
+    onClose: () => {
+      // close
+    }
+  },
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:61]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+    )
+  ]
+};
+
+export const IsCentralized: Story = {
+  args: {
+    onClose: () => {
+      // close
+    }
+  },
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:86]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentCreated/EnvironmentCreated.stories.tsx [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Default: Story = {
+  args: {}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentCreated/EnvironmentCreated.stories.tsx:20]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentNameStep/EnvironmentNameStep.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Default: Story = {
+  args: {}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentNameStep/EnvironmentNameStep.stories.tsx:20]:
```suggestion
```

File: src/components/common/icons/12px/LockIcon.tsx [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+
+const LockIconComponent = (props: IconProps) => {
+  const { size, color } = useIconProps(props);
+
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width={size}
+      height={size}
+      viewBox="0 0 10 12 "
```
Comment (kshmidt-digma) [inline src/components/common/icons/12px/LockIcon.tsx:13]:
```suggestion
      viewBox="0 0 10 12"
```

File: src/components/common/App/types.ts [hunk 24]
```diff
@@ -119,3 +121,7 @@ export interface InsightStats {
   analyticsInsightsCount: number;
   unreadInsightsCount: number;
 }
+
+export interface UserInfo {
+  id?: string;
```
Comment (kshmidt-digma) [inline src/components/common/App/types.ts:126]: Check if the same as `userId`

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypePanel/EnvironmentTypePanel.stories.tsx [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Default: Story = {
+  args: {}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypePanel/EnvironmentTypePanel.stories.tsx:22]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypePanel/EnvironmentTypePanel.stories.tsx [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+export const Centralized: Story = {
+  args: {},
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypePanel/EnvironmentTypePanel.stories.tsx:26]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypeStep/EnvironmentTypeStep.stories.tsx [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Default: Story = {
+  args: {}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/CreateEnvironmentWizard/EnvironmentTypeStep/EnvironmentTypeStep.stories.tsx:20]:
```suggestion
```

File: src/components/RecentActivity/NoData/NoData.stories.tsx [hunk 28]
```diff
@@ -0,0 +1,18 @@
...
+    layout: "fullscreen"
+  }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  args: {}
```
Comment (kshmidt-digma) [inline src/components/RecentActivity/NoData/NoData.stories.tsx:17]:
```suggestion
```

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx (hunk 18 above)
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:35]: this args is required for this component

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx (hunk 19 above)
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:61]: this args is required for this component

File: src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx (hunk 20 above)
Comment (opoliarush) [reply src/components/RecentActivity/CreateEnvironmentWizard/CreateEnvironmentWizard.stories.tsx:86]: this args is required for this component

**Review by kshmidt-digma — APPROVED** [review]

**Review by kshmidt-digma — APPROVED** [review]

**Pull request merged**